### PR TITLE
feat: KPI-201 add new metric in the investment repository

### PR DESCRIPTION
**Pull request by FreddyJR1995**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-201)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [x] Unit Test.
- [x] Coverage.

### CODE REVIEW

File: src/utils/app_names.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class ScenarioNames(StrEnum):
 class MetricNames(StrEnum):
     REVENUE = "Revenue"
     EBITDA = "Ebitda"
-    COST_OF_GOODS = "Cost of goods"
+    COST_OF_GOODS = "Cost of Goods"
```
Comment (kellycastrof) [inline src/utils/app_names.py:25]: Change this name, the variable is Cost of goods

File: src/service/comparison_vs_peers/comparison_vs_peers_service.py [hunk 2]
```diff
@@ -24,6 +24,22 @@ def remove_base_metrics(self, company: dict) -> None:
         for metric in base_metrics:
             company.pop(metric, None)
 
+    def anonymized_companies(
```
Comment (kellycastrof) [inline src/service/comparison_vs_peers/comparison_vs_peers_service.py:27]: You have a similar function on the investment year report file, could you please move it to the calendar report_

**Review by kellycastrof — CHANGES_REQUESTED** [review]

File: src/utils/app_names.py (hunk 1 above)
Comment (FreddyJR1995) [reply src/utils/app_names.py]: name changed

File: src/service/comparison_vs_peers/comparison_vs_peers_service.py (hunk 2 above)
Comment (FreddyJR1995) [reply src/service/comparison_vs_peers/comparison_vs_peers_service.py]: it was moved

File: src/service/comparison_vs_peers/comparison_vs_peers_service.py [hunk 3]
```diff
@@ -33,14 +33,15 @@ def get_comparison_vs_data(self, data: dict, access: bool) -> list:
             company_revenue = company_data.get("actuals_revenue")
             self.report.calculate_metrics(company_data)
             if not access and id not in allowed_companies:
-                self.report.anonymized_company(company_data, allowed_companies)
+                self.report.anonymize_name(company_data, allowed_companies)
```
Comment (kellycastrof) [inline src/service/comparison_vs_peers/comparison_vs_peers_service.py:36]: Why do you have the same validation inside the function and in line 35?  `id not in allowed_companies`

File: src/service/calculator/calculator_report.py [hunk 4]
```diff
@@ -104,9 +113,31 @@ def get_rule_of_40(self, company: dict, company_revenue: int) -> dict:
...
 
-    def anonymized_company(self, company: dict, allowed_companies: list) -> None:
+    def anonymized_companies_metrics(
+        self, access: bool, companies: dict, allowed_companies: list
+    ) -> None:
+        gross_profit_records = []
+
+        for id in companies:
+            company = companies[id]
+            gross_profit_records.append(company.get("gross_profit"))
```
Comment (kellycastrof) [inline src/service/calculator/calculator_report.py:123]: Use list comprehension instead
`gross_profit_records = [companies[company_id].get("gross_profit") for company_id in companies]`

File: src/service/comparison_vs_peers/comparison_vs_peers_service.py (hunk 3 above)
Comment (FreddyJR1995) [reply src/service/comparison_vs_peers/comparison_vs_peers_service.py]: I have removed the validation within the function

File: src/service/calculator/calculator_report.py (hunk 4 above)
Comment (FreddyJR1995) [reply src/service/calculator/calculator_report.py]: I changed it

File: src/service/calculator/calculator_report.py [hunk 5]
```diff
@@ -104,13 +113,34 @@ def get_rule_of_40(self, company: dict, company_revenue: int) -> dict:
             "revenue": self.calculator.calculate_base_metric(company_revenue),
         }
 
-    def anonymized_company(self, company: dict, allowed_companies: list) -> None:
+    def anonymized_companies_metrics(
+        self, access: bool, companies: dict, allowed_companies: list
+    ) -> None:
+        gross_profit_records = []
```
Comment (kellycastrof) [inline src/service/calculator/calculator_report.py:119]: Delete this line

**Review by kellycastrof — CHANGES_REQUESTED** [review]
Getting the peers report doesn't work when there are no gross profit values, could you check it out?

**Review by kellycastrof — APPROVED** [review]
Use git squash before merging
Good job

**Pull request merged**